src/controllers/queries/__tests__/index.test.ts

```diff
@@ -10,117 +10,126 @@ describe('QueriesController', () => {
   })
   describe('queries', () => {
     it('should match the snapshot of the rendered query page', async () => {
-      const { scope3CarbonConsumptionTemplateMock, mockLogger, queryTemplateMock, dbMock, queryListTemplateMock } =
-        withQueriesMocks()
-      const controller = new QueriesController(
-        scope3CarbonConsumptionTemplateMock,
-        queryTemplateMock,
-        queryListTemplateMock,
-        dbMock,
-        mockLogger
-      )
+      const { args } = withQueriesMocks()
+      const controller = new QueriesController(...args)
       const result = await controller.queries().then(toHTMLString)
       expect(result).to.equal('queries_template')
     })
     it('should call db as expected', async () => {
-      const { scope3CarbonConsumptionTemplateMock, mockLogger, queryTemplateMock, dbMock, queryListTemplateMock } =
-        withQueriesMocks()
-      const controller = new QueriesController(
-        scope3CarbonConsumptionTemplateMock,
-        queryTemplateMock,
-        queryListTemplateMock,
-        dbMock,
-        mockLogger
-      )
-      const spy = sinon.spy(dbMock, 'get')
+      const { dbMock, args } = withQueriesMocks()
+      const controller = new QueriesController(...args)
+      const spy = dbMock.get
       await controller.queryManagement().then(toHTMLString)
-      expect(spy.firstCall.calledWith('connection', {}, [['updated_at', 'desc']])).to.equal(true)
+      expect(spy.firstCall.calledWith('connection', [], [['updated_at', 'desc']])).to.equal(true)
```

Comment:
why do we  change `{}` to `[]` here but not on the query?

Reply:
ignore me ... it's because of line 51 in `[src/controllers/queries/index.ts](https://github.com/digicatapult/veritable-ui/pull/118/files#diff-b1bbe14f1e7c33853c2a15c6eb85b152df95736b040dc81d2d467ada29b2b50b)`

Reply:
on the connection we need to conditionally build up the where clause. On the query we don't. It's easier to do this with the array syntax.

The reason I changed how this was being built was because the type inference was getting things wrong. I had to add a type annotation which made the ternary very long. So I split the syntax out